va/validation-authority.go

```diff
@@ -224,8 +245,25 @@ func (va ValidationAuthorityImpl) validateDvsni(identifier core.AcmeIdentifier,
 	z := sha256.Sum256(RS)
 	zName := fmt.Sprintf("%064x.acme.invalid", z)
 
+	addrs, _, _, err := va.DNSResolver.LookupHost(identifier.Value)
+	if err != nil {
+		challenge.Status = core.StatusInvalid
+		setChallengeErrorFromDNSError(err, &challenge)
+		va.log.Debug(fmt.Sprintf("%s [%s] DNS failure: %s", challenge.Type, identifier, err))
+		return challenge, challenge.Error
+	}
+	if len(addrs) == 0 {
+		challenge.Error = &core.ProblemDetails{
+			Type:   core.UnknownHostProblem,
+			Detail: fmt.Sprintf("Could not resolve %s", identifier.Value),
+		}
+		challenge.Status = core.StatusInvalid
+		return challenge, errors.New(challenge.Error.Detail)
+	}
+	challenge.ResolvedAddrs = addrs
+
 	// Make a connection with SNI = nonceName
-	hostPort := identifier.Value + ":443"
+	hostPort := fmt.Sprintf("%s:443", addrs[0].String())
```

Comment:
I'd recommend `net.JoinHostPort()` here, in case `addrs[0]` is an IPv6 address with colons.